### ISSUE TITLE
add test for non existing document to see if CORS is correct

### ISSUE THREAD

**Issue by ghost** [remotestorage/api-test-suite#26]
My code has a different flow for exceptions (i.e. file does not exist) and the exceptions didn't add CORS headers...

**Comment by raucao** [comment]
Wait, was it a 500? Then the non-existing doc (which is tested I believe) would be a random thing to test for, if what you actually want is trigger a 500.

**Comment by ghost** [comment]
No, it was a 404. I guess it doesn't make that much sense actually to test this since a 404 is 'expected behavior'. It is just that in my implementation they follow a different program flow for which I initially forgot to add CORS. That is fixed now, so it can't happen anymore, but I thought other implementation may be similar in having a different flow for 'failures'.

**Comment by raucao** [comment]
We should totally add a test for 404s then. Very easy.

**Comment by untitaker** [comment]
What if we checked the CORS headers inside `do_put_request` et al?

**Comment by raucao** [comment]
Yes, makes sense I assume.
